lib/trie/trie.go

```diff
@@ -383,22 +394,23 @@ func (t *Trie) insertKeyLE(keyLE, value []byte,
 // insert inserts a value in the trie at the key specified.
 // It may create one or more new nodes or update an existing node.
 func (t *Trie) insert(parent *Node, key, value []byte,
-	pendingDeltas DeltaRecorder) (newParent *Node,
+	isValueHashed bool, pendingDeltas DeltaRecorder) (newParent *Node,
 	mutated bool, nodesCreated uint32, err error) {
 	if parent == nil {
 		mutated = true
 		nodesCreated = 1
 		return &Node{
-			PartialKey:   key,
-			StorageValue: value,
-			Generation:   t.generation,
-			Dirty:        true,
+			PartialKey:    key,
+			StorageValue:  value,
```

Comment:
why are we storing the value in memory even if the value is hashed and stored in the DB?

Reply:
I think we have to rethink the way we are retrieving this values. We should use lazy loading from the db but I have to check if other parts of the code are ready for that. I'm pretty sure the only part that loads the tries from DB is the [storage.go](https://github.com/ChainSafe/gossamer/blob/development/dot/state/storage.go#L103) but we are using standalone tries in different places.

Let me think more about it and find a generic solution